File: src/org/tec/datastructures/test/QueueTest.java
package org.tec.datastructures.test;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;
import org.tec.datastructures.linear.Queue;

class QueueTest {

	@Test
	public void TestEnqueue() {
		Queue<Integer> test = new Queue<Integer>();
		test.enqueue(1);
		test.enqueue(3);
		int result = test.peek();
		assertEquals(1, result);
	}
	
	@Test
	public void TestDequeue() {
		Queue<Integer> test = new Queue<Integer>();
		assertNull(test.peek());
		test.enqueue(1);
		test.enqueue(3);
		test.dequeue();
		int result = test.peek();
		assertEquals(3, result);
	}
	
	@Test
	public void TestPeek() {
		Queue<Integer> test = new Queue<Integer>();
		assertNull(test.peek());
		test.enqueue(1);
		test.enqueue(3);
		test.enqueue(6);
		test.enqueue(17);
		int result = test.peek();
		assertEquals(1, result);
	}
	
	@Test
	public void TestSize() {
		Queue<Integer> test = new Queue<Integer>();
		assertNull(test.peek());
		test.enqueue(1);
		test.enqueue(3);
		test.dequeue();
		test.enqueue(4);
		int result = test.size();
		assertEquals(2, result);
	}
	
	@Test
	public void TestPrint() {
		Queue<Integer> test = new Queue<Integer>();
		assertNull(test.peek());
		test.enqueue(1);
		test.dequeue();
		test.print();
		assertNull(test.peek());
		test.enqueue(3);
		test.print();
		int result = test.peek();
		assertEquals(3, result);
	}
	
	@Test
	public void TestEmpty() {
		Queue<Integer> test = new Queue<Integer>();
		assertNull(test.peek());
		test.enqueue(1);
		test.enqueue(3);
		assertNotNull(test.peek());
	}
	
	
	
	
	
	
	
	
	
	
	
}

File: Rider/org.tec.datastructures/Tests/QueueTest.cs
<<<<<<< HEAD
﻿namespace org.tec.datastructures.Tests
{
    public class QueueTest
    {
        
    }
=======
﻿using NUnit.Framework;
using org.tec.datastructures.Linear;

namespace org.tec.datastructures.Tests{
    
    [TestFixture()]
    public class QueueTest{
        
        [Test]
        public void testEnqueue() {
            Queue<int> tester = new Queue<int>();
            Assert.Equals(tester.Size(), 0);
            Assert.Null(tester.Peak());
            tester.Enqueue(0);
            Assert.NotNull(tester.Peak());
            tester.Enqueue(1);
            tester.Enqueue(2);
            Assert.Equals(tester.Size(), 3);
        }
    
        [Test]
        public void testDequeue() {
            Queue<int> tester = new Queue<int>();
            tester.Enqueue(0);
            tester.Enqueue(1);
            tester.Enqueue(2);
            Assert.Equals(tester.Dequeue().GetValue(), 0);
            Assert.Equals(tester.Dequeue().GetValue(), 1);
            Assert.Equals(tester.Dequeue().GetValue(), 2);
            Assert.Null(tester.Dequeue());
            Assert.Equals(tester.Size(), 0);
        }
    
        [Test]
        public void TestPeak() {
            Queue<int> tester = new Queue<int>();
            tester.Enqueue(0);
            tester.Enqueue(1);
            tester.Enqueue(2);
            Assert.Equals(tester.Size(), 3);
            Assert.Equals(tester.Peak().GetValue(), 0);
            Assert.Equals(tester.Size(), 3);
            tester.Dequeue();
            Assert.Equals(tester.Size(), 2);
            Assert.Equals(tester.Peak().GetValue(), 1);
            Assert.Equals(tester.Size(), 2);
            tester.Clear();
            Assert.Null(tester.Peak());
        }
        
    }
>>>>>>> branch 'master' of https://github.com/Greibus/Dashboard.git
}